app/code/community/Ebizmarts/MailChimp/Model/ProcessWebhook.php

```diff
@@ -272,10 +281,53 @@ public function _profile(array $data)
                      */
                     $this->_addSubscriberData($subscriber, $fname, $lname, $email, $listId);
                 }
+
+                if (isset($data['merges']['GROUPINGS'])) {
+                    $this->_processGroupsData($data['merges']['GROUPINGS'], $subscriber, true);
+                }
             }
         }
     }
 
+    protected function _processGroupsData($grouping, $data, $isDataSubscriber = false)
+    {
+        $helper = $this->getHelper();
+        $storeId = $this->_getStoreId();
+
+        $customerEmail = $data->getEmail();
+        $subscriber = $this->getSubscriberModel()->loadByEmail($customerEmail);
+
+        if (!$isDataSubscriber) {
+            $customerId = $data->getId();
+            $name = $data->getName();
+            $lastName = $data->getLastName();
+        } else {
+            $customerId = $data->getCustomerId();
+            $name = $data->getSubscriberFirstname();
+            $lastName = $data->getSubscriberLastname();
+        }
+
+        $interestGroup = $this->getInterestGroupModel();
+
+        if (!$subscriber->getSubscriberId()) {
+            $subscriber->setSubscriberEmail($customerEmail);
+            $subscriber->setSubscriberFirstname($name);
+            $subscriber->setSubscriberLastname($lastName);
+            $subscriber->subscribe($customerEmail);
+        }
```

Comment:
Why are we always subscribing without checking the status? This should only happen on **_subscribe** function.